planner/core/optimizer.go

```diff
@@ -34,6 +34,7 @@ const (
 	flagBuildKeyInfo
 	flagDecorrelate
 	flagEliminateAgg
+	flagEliminateProjection2
```

Comment:
can we move `flagEliminateProjection` to here and remove `flagEliminateProjection2`?

Reply:
No, first `project eliminate` influences the following rules. If we remove it, some cases will be less optimized.

Reply:
@winoros 
Please check https://github.com/pingcap/tidb/pull/7909.
Will this PR have bad influence?

Reply:
@winoros In #7909, `flagEliminateProjection` was moved after `flagEliminateAgg`. Why move it back?